packages/core/src/browser/frontend-application.ts

```diff
@@ -263,10 +265,10 @@ export class FrontendApplication {
             return await this.layoutRestorer.restoreLayout(this);
         } catch (error) {
             if (ApplicationShellLayoutMigrationError.is(error)) {
-                console.warn(error.message);
-                console.info('Initializing the default layout instead...');
+                this.logger.warn(error.message);
```

Comment:
is this a special case where https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#logging are not applicable?

Reply:
Thanks for that, I was wondering why it was using `console`.
However, the wiki page states these are intercepted and output in the logger. In our experience that isn't the case and this seems to be broken/never worked.

Personally I'd prefer to be explicit and use the logger, @akosyakov what do you think?

Reply:
It should work, we override console: https://github.com/eclipse-theia/theia/blob/cc51575a2c8660f13794ed9099ee14daaca982b9/packages/core/src/node/logger-backend-module.ts#L56-L61 to avoid using logger directly and log output from 3rd party extensions. We used it to use Logger directly but after figuring out that it is not available in all contexts, decided default to using `console`. It should work always everywhere.

Reply:
Updated to use `console.*` instead of `ILogger` for the entirety of `frontend-application.ts` and `backend-application.ts`.